lambdas/report-user-identity/src/main/java/uk/gov/di/ipv/core/reportuseridentity/ReportUserIdentityHandler.java

```diff
@@ -0,0 +1,328 @@
+
+            for (var page :
+                    vcStoreItemScanDynamoDataStore.getItems(
+                            reportProcessingRequest.tacticalStoreLastEvaluatedKey(),
+                            ATTR_NAME_USER_ID)) {
+
+                var reportUserIdentities =
+                        page.items().stream()
+                                .map(VcStoreItem::getUserId)
+                                .distinct()
+                                .map(
+                                        usrId ->
+                                                new ReportUserIdentityItem(
+                                                        usrId, null, null, null, null))
+                                .toList();
+                reportUserIdentityScanDynamoDataStore.createOrUpdate(reportUserIdentities);
+
+                reportProcessingResult.tacticalStoreLastEvaluatedKey(page.lastEvaluatedKey());
+
+                checkLambdaRemainingExecutionTime(context);
+            }
+            LOGGER.info(
+                    LogHelper.buildLogMessage(
+                            "Tactical storage scan completed to get unique users."));
+            reportProcessingResult.tacticalStoreLastEvaluatedKey(null);
+        }
+    }
+
+    private void processUsersToFindLOCAndUpdateDb(
+            ReportProcessingRequest reportProcessingRequest,
+            ReportProcessingResult.ReportProcessingResultBuilder reportProcessingResult,
+            Context context)
+            throws CredentialParseException, HttpResponseExceptionWithErrorBody, ParseException,
+                    BatchDeleteException, StopBeforeLambdaTimeoutException {
+        if (reportProcessingRequest.continueUserIdentityScan()) {
+            LOGGER.info(
+                    LogHelper.buildLogMessage(
+                            "Processing unique users to update their aggregate identities."));
+
+            List<ReportUserIdentityItem> totalReportUserIdentityItems = new ArrayList<>();
+            for (var page :
+                    reportUserIdentityScanDynamoDataStore.getItems(
+                            reportProcessingRequest.userIdentitylastEvaluatedKey(),
+                            ATTR_NAME_USER_ID,
+                            "identity",
+                            "migrated")) {
+                var userIds = page.items().stream().map(ReportUserIdentityItem::getUserId).toList();
+                LOGGER.info(
+                        LogHelper.buildLogMessage(
+                                String.format(
+                                        "Checking user identity for total (%s) users.",
+                                        userIds.size())));
+                List<ReportUserIdentityItem> reportUserIdentityItems = new ArrayList<>();
+                for (String userId : userIds) {
+                    evaluateUserIdentityDetail(reportUserIdentityItems, userId);
+                }
```

Comment:
Are you confident this will always evaluate within the lambda timeout (1s)? We might need to adjust the timeout to capture this _or_ move the remaining execution time check inside here.

If you're not sure, we can see how it performs in build and adjust from there.

Reply:
Not sure, will need to observe in build. Changed timeout to (1min)